tests/testthat/test-tmb.R

```diff
@@ -36,6 +36,28 @@ test_that("h_mmrm_tmb_formula_parts works as expected", {
     class = "mmrm_tmb_formula_parts"
   )
   expect_identical(result, expected)
+  expect_error(
+    h_mmrm_tmb_formula_parts(FEV1 ~ RACE + AVISIT + USUBJID),
+    paste(
+      "Covariance structure must be specified in formula.",
+      "Possible covariance structure include: us, toep, toeph, ar1, ar1h, ad, adh, cs, csh"
+    )
+  )
+  expect_error(
+    h_mmrm_tmb_formula_parts(FEV1 ~ RACE + arh1(AVISIT | USUBJID)),
+    paste(
+      "Covariance structure must be specified in formula.",
```

Comment:
In this case maybe an error pointing out that a wrong structure was specified would be even more helpful?

Reply:
I think this is a good idea, but I wonder if it will add some complexity here as function calls are possible in a formula, e.g. I have a function called `arh1`,  users may specify this term `arh1( a|b)`, where a and b are treated as logical.

In my opinion, this will add too much complexity if we are going to implement this

workflow:
1. check all terms that has a function call
2. check if these functions are available (e.g., offset, strata, these functions have special meanings in glm/cox models; arm in tern; or other built in functions like exp)
3. if functions are not available, check the similarity to our pre-defined covariance structure and report the most similar one

Reply:
by the way, this may introduce additional dependencies like `stringdist`; or we rewrite it from scratch?

Reply:
hm ok that sounds a bit too complex for the value we get from it. Then let's keep as is on this point.